dash-renderer/src/observers/prioritizedCallbacks.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    flatten,
+    includes,
+    map,
+    partition,
+    pluck,
+    reduce,
+    sort,
+    uniq
+} from 'ramda';
+
+import { IStoreState } from "../store";
+
+import {
+    aggregateCallbacks,
+    removePrioritizedCallbacks,
+    addExecutingCallbacks,
+    executeCallback
+} from '../actions/callbacks';
+
+import { stringifyId } from '../actions/dependencies';
+
+import {
+    combineIdAndProp
+} from '../actions/dependencies_ts';
+
+import isAppReady from '../actions/isAppReady';
+
+import {
+    ICallback,
+    IExecutingCallback,
+    ICallbackProperty
+} from '../types/callbacks';
+import { IStoreObserverDefinition } from '../StoreObserver';
+
+const sortPriority = (c1: ICallback, c2: ICallback): number => {
+    const lDiff = c2.priority.length - c1.priority.length;
+    if (lDiff) {
+        return lDiff;
+    }
+
+    for (let i = 0; i < c1.priority.length; ++i) {
+        const pDiff = c2.priority[i] - c1.priority[i];
+        if (pDiff) {
+            return pDiff;
+        }
```

Comment:
Following on my previous comment/question: `sortPriority` first sorts by the total number of items in this array - ie not the total number of other callbacks you're blocking, but the total number of _blocking_ callbacks you're blocking - then by number of direct blocks of this callback, followed by number of direct blocks of blocked callbacks (again, in a somewhat meaningless order). Am I understanding that correctly?

This seems like a plausible sorting, and I guess with the information at hand there probably isn't a single best answer, just trying to understand what was done and why. Unless the specific logic here is important, it feels like there may be a simpler way to do this that would also collect all the logic in one place rather than separate `getPriority` and `sortPriority` each holding part of the logic. Like, have `getPriority` just reduce to a single score and sort descending on that score.

Reply:
Priority sorting is done by first sorting by the depth of the callback tree and then by looking at the breadth of the callback tree, giving priority to a larger number of callbacks at smaller depths in the tree of a given callback.

Reply:
> Like, have getPriority just reduce to a single score and sort descending on that score.

Could probably make it a string of format `dd.dd.dd.dd` with each fragment containing the number of nodes, `0` padded and pruned at `99`. The first entry would be the depth itself. So for example `[4,3,2]` would become `04.04.03.02` and `[102,67]` would become `02.99.67`

Reply:
https://github.com/plotly/dash/pull/1254/commits/e50b7c2eff303b2093924f7903b47ad7c01f6874